tests/posix/common/src/pthread.c

```diff
@@ -465,6 +465,54 @@ ZTEST(pthread, test_pthread_cleanup)
 	zassert_ok(pthread_join(th, NULL));
 }
 
+static void *test_pthread_cancel_fn(void *arg)
+{
+	int oldtype = ~PTHREAD_CANCEL_DEFERRED;
+	int oldstate = ~PTHREAD_CANCEL_ENABLE;
+
+	zassert_ok(pthread_setcanceltype(PTHREAD_CANCEL_DEFERRED, &oldtype));
```

Comment:
why is this neccesary? the function-under-test (`pthread_testcancel`) doesnt seem to care about the type